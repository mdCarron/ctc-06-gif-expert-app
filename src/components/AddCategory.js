import React, { Fragment, useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
    console.log("handleinputchange llamado");
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategorias((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
    <Fragment>
      <form onSubmit={handleSubmit}>
        <p>{inputValue}</p>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </Fragment>
  );
};

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired,
};

export default AddCategory;
