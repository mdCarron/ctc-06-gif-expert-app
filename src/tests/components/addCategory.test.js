import React from "react";
import AddCategory from "../../components/AddCategory";
import { shallow } from "enzyme";

describe("test en componente <AddCategory />", () => {
  const setCategorias = () => {};
  const wrapper = shallow(<AddCategory setCategorias={setCategorias} />);

  test("should snapshot match", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";

    input.simulate("change", { target: { value } });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });
});
