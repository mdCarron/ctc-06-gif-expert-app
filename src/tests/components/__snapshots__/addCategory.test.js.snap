// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en componente <AddCategory /> should snapshot match 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
